mydumper/mydumper.go

```diff
@@ -168,11 +168,18 @@ func (m *Mydumper) spawn(ctx context.Context) ([]byte, error) {
 				continue
 			case "ERROR":
 				m.logger.Error(string(msg))
+				if strings.HasPrefix(string(msg), "Couldn't acquire global lock") {
+					err = terror.ErrDumpUnitGlobalLock
```

Comment:
Is it possible that this `err` overwrite by any later one, like in L184?

Reply:
https://github.com/pingcap/dm/pull/857/commits/433edcb16b8397d6026f2f273afd9168d4520588 

by set `err` function will return earlier than L184, this commit ⬆️ detect `err` in a more early place